contracts/access-control/extensions/RBACGroupable.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "../../interfaces/access-control/extensions/IRBACGroupable.sol";
+
+import "../../libs/arrays/SetHelper.sol";
+
+import "../RBAC.sol";
+
+abstract contract RBACGroupable is RBAC, IRBACGroupable {
+    using StringSet for StringSet.Set;
+    using SetHelper for StringSet.Set;
+
+    mapping(address => StringSet.Set) private _userGroups;
+    mapping(string => StringSet.Set) private _groupRoles;
+
+    function __GroupRBAC_init() internal onlyInitializing {
+        __RBAC_init();
+    }
+
+    function addUserToGroups(
+        address who_,
+        string[] memory groupsToAddTo_
+    ) public virtual override onlyPermission(RBAC_RESOURCE, CREATE_PERMISSION) {
+        require(groupsToAddTo_.length > 0, "RBACGroupable: empty groups");
+
+        _addUserToGroups(who_, groupsToAddTo_);
+    }
+
+    function removeUserFromGroups(
+        address who_,
+        string[] memory groupsToRemoveFrom_
+    ) public virtual override onlyPermission(RBAC_RESOURCE, DELETE_PERMISSION) {
+        require(groupsToRemoveFrom_.length > 0, "RBACGroupable: empty groups");
+
+        _removeUserFromGroups(who_, groupsToRemoveFrom_);
+    }
+
+    function grantGroupRoles(
+        string calldata groupTo_,
```

Comment:
Please use `memory` in the contract. The interface is fine

Reply:
done